Add 3D physics interpolation demo

Calinou opened the pull request:
Demonstrate first person shooter and third person shooter cameras.

This salvages @lawnjelly's work on the demo and further refines it.

- This closes https://github.com/godotengine/godot-demo-projects/pull/1220 by superseding it.

## Preview

![image](https://github.com/user-attachments/assets/dfc592ed-2dc4-4696-946a-a9bae751ec69)

lawnjelly commented:
This is great! :partying_face: 

I've added a bunch more comments and made some variable names clearer, hopefully you can copy the gdscript from:
https://github.com/godotengine/godot-demo-projects/pull/1220

I also discovered a bug in the `SceneTreeFTI` for toggling interpolation on / off :blush: , but have made a small fix PR so hopefully that will be closed for 4.5.

Calinou commented:
> I've added a bunch more comments and made some variable names clearer, hopefully you can copy the gdscript from:
https://github.com/godotengine/godot-demo-projects/pull/1220

Done. Note that I've refactored `bullet.gd` to use built-in Godot functionality whenever possible (and also removed the scale clamping, as current Godot should have no problems with zero-scaled meshses).

lawnjelly commented on 3d/physics_interpolation/player.gd:
```diff
@@ -0,0 +1,202 @@
+func _update_camera(delta: float) -> void:
+	# Keep the player direction up to date based on the yaw.
+	_dir.x = sin(_yaw)
+	_dir.z = cos(_yaw)
```
Note that I flipped the polarity of `_dir` in my changes, as these actually point behind the player rather than ahead. It's okay doing it this way round too but you might need to change the comment later on where I mentioned flipping the polarity, which now no longer matches up with the code.
